Спринт №8

Keithcutt opened the pull request:
1. Свёрстаны главный экран и экран настроек
2. Реализованы нажатия на кнопки главного экрана и кнопку "назад" экрана настроек

Keithcutt committed:
Выполненное домашнее задание 8 спринта:
1. Свёрстаны главный экран и экран настроек
2. Реализованы нажатия на кнопки главного экрана и кнопку "назад" экрана настроек.

ArturNurtdinov commented on app/src/main/res/drawable/ic_share.xml:
```diff
@@ -0,0 +1,9 @@
+    android:viewportHeight="24">
+  <path
+      android:pathData="M17.333,15.723C16.658,15.723 16.053,15.994 15.591,16.419L9.253,12.669C9.298,12.461 9.333,12.253 9.333,12.036C9.333,11.819 9.298,11.611 9.253,11.404L15.52,7.69C16,8.142 16.631,8.422 17.333,8.422C18.809,8.422 20,7.211 20,5.711C20,4.211 18.809,3 17.333,3C15.858,3 14.667,4.211 14.667,5.711C14.667,5.928 14.702,6.136 14.747,6.343L8.48,10.057C8,9.605 7.369,9.325 6.667,9.325C5.191 ...
+      android:fillColor="#AEAFB4"/>
```
В иконках вместо `fillColor` ты можешь подставлять ссылку на цвет из палитры, например:
```suggestion
      android:fillColor="@color/YP_text_gray"/>
```
В будущем это пригодится, когда добавится поддержка тёмной темы, иконки сами будут адаптироваться под тему и не нужно будет создавать новые иконки для тёмной темы

ArturNurtdinov commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,90 @@
+        android:textSize="@dimen/heading_text"
+        android:textColor="@color/YP_black"
+        android:textAllCaps="false"
+        android:textStyle="bold" >
```
Судя по дизайну, стиль текста - normal
Ты можешь добавить шрифты в приложение и использовать шрифт medium, либо оставить просто normal стиль текста с обычным шрифтом

ArturNurtdinov commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,148 @@
+            android:layout_gravity="end"
+            android:gravity="center_vertical"
+            android:paddingVertical="11dp"
+            android:paddingStart="9dp"
```
В дизайне могут быть нечетные размеры, или не симметричные. Но лучше использовать в коде размеры, кратные 4 или 8. Документация - https://material.io/design/layout/spacing-methods.html#baseline-grid

ArturNurtdinov commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,148 @@
+        android:layout_height="@dimen/settings_options_height"
+        android:layout_marginTop="@dimen/first_element_margin_top"
+        android:paddingStart="@dimen/padding_standard"
+        android:paddingEnd="6dp"
```
Лучше все размеры выносить в dimens, не только те, которые несколько раз используются в вёрстке. К тому же, для удобства ты можешь заводить dimen с именами `dp_6` или `dp_24`, чтобы не придумывать каждый раз новые имена для очередного отступа или размера и не дублировать это в dimens.xml

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,11 +1,35 @@
+
+        val searchButton = findViewById<MaterialButton>(R.id.search_btn)
+        val mediaButton = findViewById<MaterialButton>(R.id.media_btn)
+        val settingsButton = findViewById<MaterialButton>(R.id.settings_btn)
```
переменные рекомендуется объявлять максимально близко к месту их использования, это облегчает чтение кода

Keithcutt committed:
Исправления после ревью (8 спринт):

1. Вместо атрибута fillColor у векторных изображений подставлены ссылки на цвета из colors.
2. Убран стиль заголовков обоих экранов bold -> normal.
3. Скорректированы нечетные размеры отступов.
4. Все размеры вынесены в dimens, дублирующиеся размеры убраны или переименованы (в формате "dp_6").

Keithcutt committed:
Еще одно исправление (8 спринт):

1. Проанализировав последний коммит понял, что ошибся в размерах padding_standart в файле activity_main. Исправил размеры на корректные (12dp -> 16dp).
